Support parameterized lazy loading DataProvider

krmahadevan opened the pull request:
Closes #2565

Fixes #2565  .

### Did you remember to?

- [X] Add test case(s)
- [X] Update `CHANGES.txt`

We encourage pull requests that:

* Add new features to TestNG (or)
* Fix bugs in TestNG

If your pull request involves fixing SonarQube issues then we would suggest that you please discuss this with the 
[TestNG-dev](https://groups.google.com/forum/#!forum/testng-dev) before you spend time working on it.

juherr commented on core/src/test/java/test/dataprovider/issue2565/SampleTestUsingSupplier.java:
```diff
@@ -0,0 +1,38 @@
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+public class SampleTestClass {
```
The class name doesn't match the file name: `SampleTestUsingSupplier`

krmahadevan replied:
Dang! That was a bad commit. Missed adding the refactored class into the commit. This is now fixed.

krmahadevan commented:
@juherr - Please let me know if there is anything else that needs to be fixed in this PR. I have rebased off of master and fixed merge conflicts as well.
